clevertap-core/src/main/java/com/clevertap/android/sdk/inapp/CTInAppAction.kt

```diff
@@ -0,0 +1,120 @@
+package com.clevertap.android.sdk.inapp
+
+import android.os.Parcel
+import android.os.Parcelable
+import android.os.Parcelable.Creator
+import com.clevertap.android.sdk.Constants
+import com.clevertap.android.sdk.Constants.KEY_FALLBACK_NOTIFICATION_SETTINGS
+import com.clevertap.android.sdk.inapp.InAppActionType.OPEN_URL
+import com.clevertap.android.sdk.inapp.customtemplates.CustomTemplateInAppData
+import com.clevertap.android.sdk.utils.getStringOrNull
+import org.json.JSONObject
+
+internal class CTInAppAction private constructor(parcel: Parcel?) : Parcelable {
+
+    var type: InAppActionType?
+        private set
+
+    var actionUrl: String?
+        private set
+
+    var keyValues: HashMap<String, String>?
+        private set
+
+    var customTemplateInAppData: CustomTemplateInAppData?
+        private set
+
+    @get:JvmName("shouldFallbackToSettings")
+    var shouldFallbackToSettings: Boolean = false
+        private set
+
+    init {
+        type = parcel?.readString()?.let { InAppActionType.fromString(it) }
+        actionUrl = parcel?.readString()
+        keyValues = parcel?.readHashMap(null) as? HashMap<String, String>
```

Comment:
readHashMap is deprecated when i checked in IDE.

Reply:
It is deprecated, but this was the way it was previously stored/read in the parcel. We could replace it with readBundle in a separate task in order to be tested properly.